### DISCUSSION TITLE
i3 workspaces display has unexpected extra padding on the left side of unfocused workspaces

### DISCUSSION THREAD

**Discussion by aaannestad** [polybar/polybar#3070] [Q&A]
I've got a very very basic polybar config I'm trying to build out, having just switched to i3wm, and I'm having very odd behaviour - namely, when I use the i3 module's workspace list, cycling between workspaces causes the spacing between the numbers to change, shifting them all over until I cycle back. Some testing and investigation seems to show that this is because unfocused workspace labels / UI boxes have about one 10pt character's worth of extra padding only on their left side - asymmetrical with their right side, and not present with focused labels/boxes.

The config is this:
```
[colours]
background = #000000
highlight = #b8bb26

[bar/test]

bottom = true

modules-center = i3
modules-right = pulseaudio

[module/i3]
type = internal/i3
pin-workspaces = true

label-focused = %index%
label-focused-foreground = ${colours.highlight}

[module/pulseaudio]
type = internal/pulseaudio
click-right = pavucontrol
```
When I start polybar the workspace list looks like this:
![image](https://github.com/polybar/polybar/assets/91560337/14ac881a-69cc-4fe2-97a9-a43203318849)
(workspace 1 is on my non-primary monitor ATM)

When I cycle to the next workspace, this happens:
![image](https://github.com/polybar/polybar/assets/91560337/549f39cc-4fdc-4312-a76a-e0dd27cbca90)
You can see that the focused workspace label is much closer to the left unfocused workspace than the right one.

With the next one, the same thing happens:
![image](https://github.com/polybar/polybar/assets/91560337/9d606126-e776-4700-abe8-2f888f9b4798)

Note also that when a workspace becomes unfocused, the label jumps to the right because of the additional unexpected padding.

Effectively, this is the situation:

c = 1 character width or so (at 10pt size at least)
p = padding specified
L = label width

Focused box size = p + L + p
Unfocused box size = p + ***c*** + L + p

In short, only the label for an unfocused box has about one extra character's width of padding on only the left side. I don't know how to either get rid of this or compensate for it, since it's not clear that there's a way to add padding only on the right side; it persists at about the same size regardless of how much additional padding I add to either the focused box or the unfocused box. Is there any way I can get rid of this unexpected additional padding?

**Comment by aaannestad** [comment]
Aha! The issue is that I hadn't defined `label-unfocused` as anything in particular, so it was defaulting to something more than just `%index%`. `label-unfocused = %index%` solved the issue.
